addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+            disconnect();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        exit = true;
+        disconnect();
+    }
+
+    private void connect() throws IOException, ConfigurationError {
+        // Note: connect may leave the port open even if it throws an exception.
+        if (serialPort == null) {
+            String portName = (String) getThing().getConfiguration().get("port");
+            if (portName == null) {
+                throw new ConfigurationError("Serial port name not configured");
+            }
+            try {
+                serialPort = new RXTXPort(portName);
+            } catch (PortInUseException e) {
+                throw new IOException(e);
+            }
+            try {
+                serialPort.setSerialPortParams(BAUD, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                        SerialPort.PARITY_NONE);
+            } catch (UnsupportedCommOperationException e) {
+                serialPort.close();
+                throw new IOException(e);
+            }
+
+            // Don't need continuous updates of the display, we still get updates when
+            // the volume, etc., changes
+            serialPort.getOutputStream().write("display_update_manual!".getBytes(StandardCharsets.US_ASCII));
+            updateStatus(ThingStatus.ONLINE);
+            serialPort.getOutputStream().write("get_current_power!".getBytes(StandardCharsets.US_ASCII));
+            updateState(CHANNEL_MUTE, OnOffType.OFF);
+            updateState(CHANNEL_BRIGHTNESS, new PercentType(100));
+        }
+    }
+
+    private void disconnect() {
+        if (serialPort != null) {
+            serialPort.close();
+        }
+        serialPort = null;
+    }
+
+    private String readCommand() throws IOException {
+        return readUntil('!', '=');
+    }
+
+    private String readUntil(char terminator) throws IOException {
+        return readUntil(terminator, terminator);
+    }
+
+    private String readUntil(char terminator1, char terminator2) throws IOException {
+        StringBuilder result = new StringBuilder();
+        while (true) {
+            int b = serialPort.getInputStream().read();
+            if (b == -1) {
+                throw new IOException("Connection unexpectedly closed");
+            } else if (b == terminator1 || b == terminator2) {
+                break;
+            } else {
+                result.append((char) b);
+            }
+        }
+        return result.toString();
+    }
+
+    private PercentType readVolume() throws IOException {
+        String volumeString = readUntil('!');
+        int volume;
+        if ("min".equals(volumeString)) {
+            volume = 0;
+        } else if ("max".equals(volumeString)) {
+            volume = maximumVolume;
+        } else {
+            volume = Integer.parseInt(volumeString, 10);
+        }
+        double volumePct = volume * 100.0 / maximumVolume;
```

Comment:
Because the `maxmumVolume` is a config parameter you probably should make sure it's never 0 here. Just to be on the save side (Maybe do a `Math.min` when reading the value from config`, I know you set the min value at 1 in channel config, but i'm not sure if that is sufficient`)